platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/codecs/FileCollectionCodec.kt

```diff
@@ -57,9 +60,16 @@ class FileCollectionCodec(
     }
 
     suspend fun WriteContext.encodeContents(value: FileCollectionInternal) {
-        val visitor = CollectingVisitor()
-        value.visitStructure(visitor)
-        writeCollection(visitor.elements)
+        val executionTimeValue = value.calculateExecutionTimeValue().getOrNull()
+        if (executionTimeValue != null) {
+            writeBoolean(true)
+            write(executionTimeValue)
+        } else {
+            writeBoolean(false)
+            val visitor = CollectingVisitor()
```

Comment:
💅 I'd extract these 3 lines to a separate method (something like `encodeFileCollectionStructure` or `encodeViaCollectingVisitor`)